packages/itwin/grouping-mapping-widget/src/widget/components/GroupAction.tsx

```diff
@@ -58,14 +75,18 @@ const GroupAction = ({
   const [validator, showValidationMessage] = useValidator();
   const [isLoading, setIsLoading] = useState<boolean>(false);
   const [isRendering, setIsRendering] = useState<boolean>(false);
-  const [currentPropertyList, setCurrentPropertyList] = React.useState<PropertyRecord[]>([]);
+  const [currentPropertyList, setCurrentPropertyList] = React.useState<
+    PropertyRecord[]
+  >([]);
   const [queryBuilder, setQueryBuilder] = React.useState<QueryBuilder>(
     new QueryBuilder(undefined),
   );
-  const [queryGenerationType, setQueryGenerationType] = React.useState("Selection");
-  const [searchInput, setSearchInput] = React.useState("");
+  const [localQueryGenerationType, setQueryGenerationType] =
```

Comment:
I would prefer having the state variable as `queryGenerationType`, since usually the state variable and setter are of the pattern `state` and `setState`. I would drop the destructuring of props from parameters of this component in favor of accessing those properties via object, e.g. `props.queryGenerationType`.

Reply:
resolved